tests/components/tessie/test_lock.py

```diff
@@ -72,8 +72,29 @@ async def test_locks(
         assert hass.states.get(entity_id).state == STATE_UNLOCKED
         mock_run.assert_called_once()
 
+
+async def test_speed_limit_lock(
+    hass: HomeAssistant, entity_registry: er.EntityRegistry
+) -> None:
+    """Tests that the lock entity is correct."""
+
+    # Create the deprecated speed limit lock entity
+    entity_id = entity_registry.async_get_or_create(
+        LOCK_DOMAIN,
+        DOMAIN,
+        "VINVINVIN-vehicle_state_speed_limit_mode_active",
+        original_name="Charge cable lock",
+        has_entity_name=True,
+        translation_key="vehicle_state_speed_limit_mode_active",
+    )
+
```

Comment:
```suggestion
    entity_id = entity.entity_id

```
Probably cleaner if you update the service calls to `entity.entity_id` instead than this.